tests/test_pipeline_mixin.py

```diff
@@ -685,3 +704,87 @@ def validate_test_components(test_case, task, model, tokenizer, processor):
             raise ValueError(
                 "Could not determine `vocab_size` from model configuration while `tokenizer` is not `None`."
             )
+
+
+def get_arg_names_from_hub_spec(hub_spec, first_level=True):
+    # This util is used in pipeline tests, to verify that a pipeline's documented arguments
+    # match the Hub specification for that task
+    arg_names = []
+    for field in fields(hub_spec):
+        # First, recurse into nested fields
+        if first_level and isclass(field.type) and issubclass(field.type, inference_specs.BaseInferenceType):
+            arg_names.extend(get_arg_names_from_hub_spec(field.type, first_level=False))
+            continue
+        # Next, catch nested fields that are part of a Union[], which is usually caused by Optional[]
+        for param_type in get_args(field.type):
+            if first_level and isclass(param_type) and issubclass(param_type, inference_specs.BaseInferenceType):
+                arg_names.extend(
+                    get_arg_names_from_hub_spec(param_type, first_level=False)
+                )  # Recurse into nested fields
+                break
+        else:
+            # Finally, this line triggers if it's not a nested field
+            arg_names.append(field.name)
+    return arg_names
+
+
+def parse_google_format_docstring_by_indentation(docstring):
```

Comment:
(nit) I'd rename this method

```
def parse_args_from_docstring(...)
```

or

```
def parse_args_from_google_format_docstring(...)
```
to clearly state that the output is the args. Btw it's a very nice method that we could reuse in `huggingface_hub` as well to check if methods are correctly documented!

Reply:
Done!